### PR TITLE
Details about using CredSSP in Windows

### PR DESCRIPTION
**Pull request by bostko**
@aledsage can you review.

### CODE REVIEW

File: guide/yaml/winrm/index.md [hunk 1]
```diff
@@ -289,11 +289,24 @@ When a script is run over WinRM, the credentials under which the script are run
...
-it is assumed that the VM is already correctly configured. Please ensure that Brooklyn's changes 
-are compatible with your organisation's security policy.
+The WinRM client uses Negotiate+NTLM to authenticate against the machine.
+This mechanism applies certain restrictions to executing commands on the windows host.
+
+For this reason you should enable CredSSP on the windows host which grants all privileges available to the user.
+ https://technet.microsoft.com/en-us/library/hh849719.aspx#sectionSection4
+
+To use `Invoke-Command -Authentication CredSSP` the Windows Machine has to have:
+- Up and running WinRM over http
```
Comment (aledsage) [inline guide/yaml/winrm/index.md]: or https?

Comment (bostko) [inline guide/yaml/winrm/index.md]: http as well

Comment (aledsage) [inline guide/yaml/winrm/index.md]: So can you change the wording so that it does not imply that it must be http?

Comment (bostko) [inline guide/yaml/winrm/index.md]: I am saying that is must have http enabled.

Comment (cloudsoftreleaser) [inline guide/yaml/winrm/index.md]: @bostko I thought we were using https with winrm4j now. Are we using http just for this? So do we enable both? Or just one of them?

In separate communication, you said about an error for MSSQL@Azure: `NoSuchElementException: could not connect to any socket in [40.68.36.154:5986, 100.89.232.98:5986]`. That is the https port, so I assumed we were using https.

**Comment by aledsage** [comment]
@neykov any comments from you about the need for http and https when we're using `Invoke-Command -Authentication CredSSP`?

File: guide/yaml/winrm/index.md [hunk 2]
```diff
@@ -289,11 +289,25 @@ When a script is run over WinRM, the credentials under which the script are run
...
+- Up and running WinRM over http. Notice that we support winrm over https but for Invoke-Command to work it needs up and running winrm over http.
+  Apache Brooklyn can winrm over https but if the install script has inside it `Invoke-Command -Authentication CredSSP` then winrm over http has to be enabled as well.
+- Added trusted host entries which will use Invoke-Command
+- Allowed CredSSP
+
+All the above requirements are enabled in Apache Brooklyn through [brooklyn-server/software/base/src/main/resources/org/apache/brooklyn/software/base/custom-enable-credssp.ps1](https://github.com/apache/brooklyn-server/blob/master/software/base/src/main/resources/org/apache/brooklyn/software/base/custom-enable-credssp.ps1)
+script which enables executing commands with CredSSP in the general case.
+The script works for most of the Windows images out there version 2008 and later.
+
+Please ensure that Brooklyn's changes are compatible with your organisation's security policy.
```
Comment (ahgittin) [inline guide/yaml/winrm/index.md]: [nit] US English `s/s/z/`

**Comment by ahgittin** [comment]
useful.  @neykov if you have any comments on http(s) (re aled's comment above) it can be a new PR

**Comment by neykov** [comment]
@aledsage Invoke-Command will default to http transport (the authentication protocol doesn't matter). The script can override it, but then needs to know what settings to use (what's available on the machine).
So the solution is to either enable http for loopback or somehow figure out what's available and use that. 

@bostko  can you confirm

Also the self signed certificate might be a problem for `Invoke-Command` - should be fine for localhost where it's already trusted, but for other cases need to use `Enter-PSSession` beforehand with ignore option.

**Comment by bostko** [comment]
@neykov indeed. The script does what you wrote.
It enables winrm and adds machine's ip and localhost to trusted remote hosts.

**Pull request merged**